lightning/src/onion_message/packet.rs

```diff
@@ -136,6 +136,20 @@ impl<T: CustomOnionMessageContents> OnionMessageContents<T> {
 	}
 }
 
+use std::fmt;
+impl<T: CustomOnionMessageContents> fmt::Display for OnionMessageContents<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            OnionMessageContents::Offers(msg) => {
+                write!(f, " (Offers): {:?}", msg)
```

Comment:
Could you implement `Display` for the underlying offers structs as well? As-is, this will log e.g. an invoice request as: 

`InvoiceRequest { bytes: [0, 32, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 8, 2, 3, 232, 10, 3, 102, 111, 111, 22, 33, 2, 187, 88, 181, 254, 202, 80, 92, 116, 237, 192, 0, 216, 40, 47, 197, 86, 229, 26, 16, 36, 252, 142, 125, 126, 86, 198, 248, 135, 197, 200, 213, 242, 88, 33, 3, 91, 229, 233, 71, 130, 9, 103, 74, 150, 230, 15, 31, 3, 127, 97, 118, 84, 15, 208, 1, 250, 29, 100, 105, 71, 112, 197, 106, 119, 9, 196, 44, 240, 64, 247, 104, 21, 70, 94, 4, 223, 193, 178, 211, 248, 77, 147, 62, 69, 78, 149, 158, 232, 138, 228, 115, 153, 143, 135, 72, 52, 163, 167, 41, 61, 196, 36, 135, 241, 186, 2, 180, 239, 239, 132, 44, 247, 229, 43, 226, 102, 210, 27, 72, 91, 222, 112, 28, 79, 246, 162, 250, 29, 99, 125, 185, 229, 236], contents: InvoiceRequestContents { inner: InvoiceRequestContentsWithoutPayerId { payer: PayerContents([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]), offer: OfferContents { chains: None, metadata: None, amount: Some(Bitcoin { amount_msats: 1000 }), description: "foo", features: [], absolute_expiry: None, issuer: None, paths: None, supported_quantity: One, signing_pubkey: PublicKey(f2d5c8c587f8c6567e7d8efc24101ae556c52f28d800c0ed745c50cafeb558bbf856735f089714075023eef9d31c8e3f13557ee6c819af9124ac578a506ff702) }, chain: None, amount_msats: None, features: [], quantity: None, payer_note: None }, payer_id: PublicKey(2cc409776ac5704769641dfa01d00f5476617f031f0fe6964a67098247e9e55bbfb8078fbda6f6c993e1f8321eaae171e1cc261ec72cd5108bbf3d84c3885a03) }, signature: Signature(f76815465e04dfc1b2d3f84d933e454e959ee88ae473998f874834a3a7293dc42487f1ba02b4efef842cf7e52be266d21b485bde701c4ff6a2fa1d637db9e5ec) }`

Which is pretty ugly, we probably only want to log a few of those fields and def not the bytes/signature.

Reply:
which of these fields want to log.

Reply:
Hmm... we may want to use `Debug` instead of `Display` -- as we do for wire messages -- given the `Offer` and `Refund` messages use the latter for bech32 encoding. We could avoid `bytes` by logging the `Debug` of each message's `as_tlv_stream()`, though it would require changing those methods and returned types to `pub(crate)`. Then we don't need to worry about ever needing to update `Display` if the internal structure of a message changes.

Reply:
I'm a bit more inclined to use the TLV version as it's more LDK-agnostic when we may need to debug messages sent from other implementations. So having the output closer to the spec may be more desirable than seeing our internal mapping.

Additionally, changing `Debug` is a bit of maintenance burden and could be error prone (i.e., dealing with multiple structs, possibly leaving out fields, etc.).